src/dune_file.ml

```diff
@@ -838,8 +823,8 @@ module Library = struct
          field "ppx_runtime_libraries" (list (located Lib_name.decode)) ~default:[]
        and c_flags = field_oslu "c_flags"
        and cxx_flags = field_oslu "cxx_flags"
-       and c_names = field "c_names" (list c_name) ~default:[]
-       and cxx_names = field "cxx_names" (list cxx_name) ~default:[]
+       and c_names = field_o "c_names" Ordered_set_lang.decode
```

Comment:
Do we need to put that behind a language version check?

Reply:
I think this change is backwards compatible.

Reply:
I suppose that if we are using (lang dune <1.7), then we should make sure the arguments are plain lists.